### PR TITLE
Example mser

### PR DESCRIPTION
**Pull request by LaurentBerger**
A new example how to use features2D for MSER.Data results are visualized in 3D using opengl with mouse or keyboard interaction

### CODE REVIEW

**Comment by vpisarev** [comment]
thank you! looks cool! :+1:

**Comment by LaurentBerger** [comment]
@vpisarev I need help in this example for linux. I use include file cvconfig.h In this file  I can ifdef macro with HAVE_OPENGL to enable some part of the code.
For linux64 it seems that cvconfig.h cannot be found. how can I know if opengl is used in opencv?

**Comment by vpisarev** [comment]
@LaurentBerger, thanks! we are working on this problem

File: samples/cpp/detect_mser.cpp [hunk 1]
```diff
@@ -0,0 +1,535 @@
+#include <opencv2/opencv.hpp>
+#include "opencv2/core/opengl.hpp"
+#include "cvconfig.h"
```
Comment (mshabunin) [inline samples/cpp/detect_mser.cpp]: @LaurentBerger, try to replace it with "opencv2/cvconfig.h".

**Comment by vpisarev** [comment]
@LaurentBerger, @mshabunin, looks like something is still wrong

**Comment by mshabunin** [comment]
The cvconfig.h file is placed to `CMAKE_BINARY_DIR` during OpenCV build and is installed to `${OPENCV_INCLUDE_INSTALL_PATH}/opencv2` folder.

We can place this file to `${CMAKE_BINARY_DIR}/opencv2` folder too, then including it as "opencv2/cvconfig.h" will work for both OpenCV and cmake test builds.

Following line should be added to `cmake/OpenCVGenHeaders.cmake`:
```.diff
--- a/cmake/OpenCVGenHeaders.cmake
+++ b/cmake/OpenCVGenHeaders.cmake
@@ -1,5 +1,6 @@
 # platform-specific config file
 configure_file("${OpenCV_SOURCE_DIR}/cmake/templates/cvconfig.h.in" "${OPENCV_CONFIG_FILE_INCLUDE_DIR}/cvconfig.h")
+configure_file("${OpenCV_SOURCE_DIR}/cmake/templates/cvconfig.h.in" "${OPENCV_CONFIG_FILE_INCLUDE_DIR}/opencv2/cvconfig.h")
```

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**